backends/intel_gpu/kernels/cross_entropy_with_softmax_kernel.cc

```diff
@@ -0,0 +1,455 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "dnn_support.hpp"
+#include "kernels.h"
+#include "paddle/phi/capi/all.h"
+#include "phi_funcs.h"
+
+namespace custom_kernel {
+
+template <typename T, typename U>
+void CrossEntropy(const T* prob,
+                  const U* label,
+                  bool soft_label,
+                  size_t batch_size,
+                  size_t num_classes,
+                  int ignore_index,
+                  int axis_dim,
+                  T* out, sycl::queue* q = nullptr) {
+  auto num_remain = num_classes / axis_dim;
+  if (soft_label) {
+    for (auto i = 0; i < batch_size; ++i) {
+      for (auto k = 0; k < num_remain; ++k) {
+        out[i * num_remain + k] = 0;
+        for (auto j = 0; j < axis_dim; ++j) {
+          auto idx = i * num_classes + j * num_remain + k;
+          out[i * num_remain + k] -=
+              label[idx] * phi::TolerableValue<T>(std::log(prob[idx]));
+        }
+      }
+    }
+  } else {
+      q->parallel_for(batch_size, [=](auto& i){
+        for (int j = 0; j < num_remain; j++) {
+          int lbl = static_cast<int>(label[i * num_remain + j]);
+          if (lbl != ignore_index) {
```

Comment:
this "if" do nothing? If yes, please remove

Reply:
fixed